test/e2e/annotation_test.go

```diff
@@ -0,0 +1,83 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	pgsqlapis "github.com/operator-backing-service-samples/postgresql-operator/pkg/apis"
+	pgv1alpha1 "github.com/operator-backing-service-samples/postgresql-operator/pkg/apis/postgresql/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/apis"
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/test/mocks"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+func TestAnnoationBasedMetadata(t *testing.T) {
+	sbrlist := v1alpha1.ServiceBindingRequestList{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, &sbrlist)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+
+	dbList := pgv1alpha1.DatabaseList{}
+	require.Nil(t, framework.AddToFrameworkScheme(pgsqlapis.AddToScheme, &dbList))
+
+	dpList := appsv1.DeploymentList{}
+	require.Nil(t, framework.AddToFrameworkScheme(appsv1.AddToScheme, &dpList))
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: retryInterval})
+	if err != nil {
+		t.Fatalf("failed to initialize cluster resources: %v", err)
+	}
+
+	// get namespace
+	namespace, err := ctx.GetNamespace()
```

Comment:
@baijum it should only be needed in e2e tests, not during unit-tests.

Reply:
@otaviof This is not a unit test, but e2e, right?